tee-worker/service/src/main.rs

```diff
@@ -441,15 +444,59 @@ fn start_worker<E, T, D, InitializationHandler, WorkerModeProvider>(
 		return
 	}
 
-	println!("[>] Register the enclave (send the extrinsic)");
-	let register_enclave_xt_hash = node_api.send_extrinsic(xthex, XtStatus::Finalized).unwrap();
-	println!("[<] Extrinsic got finalized. Hash: {:?}\n", register_enclave_xt_hash);
-
-	let register_enclave_xt_header =
-		node_api.get_header(register_enclave_xt_hash).unwrap().unwrap();
+	let mut register_enclave_xt_header: Option<Header> = None;
+	let mut we_are_primary_validateer: bool = false;
+
+	// litentry, Check if the enclave is already registered
+	match node_api.get_keys(storage_key("Teerex", "EnclaveRegistry"), None) {
+		Ok(Some(keys)) => {
+			let trusted_url = trusted_url.as_bytes().to_vec();
+			let mrenclave = mrenclave.to_vec();
+			let mut found = false;
+			for key in keys {
+				let key = if key.starts_with("0x") {
+					let bytes = &key.as_bytes()[b"0x".len()..];
+					hex::decode(bytes).unwrap()
+				} else {
+					hex::decode(key.as_bytes()).unwrap()
+				};
+				match node_api.get_storage_by_key_hash::<TeerexEnclave<AccountId32, Vec<u8>>>(
+					StorageKey(key.clone()),
+					None,
+				) {
+					Ok(Some(value)) => {
+						if value.mr_enclave.to_vec() == mrenclave && value.url == trusted_url {
```

Comment:
It's almost guaranteed that we'll fall into the `!found` branch if we are worker-2, as the `trusted_url` is different. I don't understand why we still need to register the enclave in this case, but it's the same logic as the original code